File: api/internals/services/auth/sign_in.go
package authService

import (
	"photokeep-api/api/rest/schemas"
	"photokeep-api/internals/dto"
)

func SignIn(data *schemas.Login) (*dto.SignInResponseDTO, error) {
	user, err := ValidateUser(data.Username, data.Password)
	
	if err != nil {
		return nil, err
	}
	
	session, err := CreateSession(user.Username)

	if err != nil {
		return nil, err
	}
	
	accessToken, err := CreateAccessToken(session.RefreshToken)
	
	if err != nil {
		return nil, err
	}
	
	response := &dto.SignInResponseDTO{
		AccessToken:  accessToken,
		RefreshToken: session.RefreshToken,
		User:         user,
	}
	
	return response, nil
}

File: api/src/services/auth/sign-in.ts
import type { SignInSchemaBody } from '@functions/auth/sign-in/schema'
import { signInResponseDto } from '@dto/auth'
import { findConfigByUsername } from '@repository/config'
import validateUser from './validate-user'
import createSession from './create-session'
import createAccessToken from './create-access-token'

export default async (signInSchema: SignInSchemaBody) => {
  const user = await validateUser(signInSchema.username, signInSchema.password)
  const session = await createSession(user)
  const accessToken = await createAccessToken(session.refreshToken)

  const userConfig = await findConfigByUsername(user.username)

  return signInResponseDto({
    accessToken,
    refreshToken: session.refreshToken,
    userConfig,
    user,
  })
}
